packages/climate-ref/src/climate_ref/executor/hpc.py

```diff
@@ -0,0 +1,178 @@
+        else:
+            partition_limits = None
+            node_info = None
+
+        if qos is not None:
+            if not slurm_checker.get_partition_info(qos):
+                raise ValueError(f"QOS: {qos} not valid")
+
+            if not slurm_checker.can_account_use_qos(
+                account, 
+                qos):
+                raise ValueError(f"Account: {account} cannot access qos: {qos}")
+            qos_limits =  slurm_checker.get_qos_limits(qos)
+        else:
+            qos_limits = None
+
+
+
+        max_cores_per_node = float(node_info["cpus"]) if node_info else 64
+
+        max_walltime_partition = partition_limits["max_time_minutes"] if partition_limits else walltime
+        max_walltime_qos = qos_limits["max_time_minutes"] if qos_limits else walltime
+
+        max_walltime_minutes  = min(max_walltime_partition, max_walltime_qos)
+
+        hours, minutes, seconds = map(int, walltime.split(':'))
+        total_minutes = hours * 60 + minutes + seconds / 60
+
+        if total_minutes > float(max_walltime_minutes):
+            raise ValueError(f"walltime: {walltime} exceed the maximum time " + 
+                             f"{max_walltime_minutes} allowed by {partition} and {qos}")
+
+
+        slurm_config = ParslConfig(
+            executors=[
+                HighThroughputExecutor(
+                    label = 'ref_hpc_executor',
+                    cores_per_worker = max_cores_per_node,
+                    provider=SlurmProvider(
+                        account = self.account, 
+                        partition = self.partition,
+                        qos = self.qos,
+                        nodes_per_block = self.nodes,
+                        scheduler_options = '#SBATCH -C cpu',
+                        worker_init='uv .venv/bin/activate',
+                        launcher=SrunLauncher(),
+                        walltime=self.walltime,
+                        cmd_timeout=120,
+                    ),
+                )
+            ]
+        )
+        parsl.load(slurm_config)
+   
+        self.parsl_results: list[Any] = []
+
+
+    def run(
+        self,
+        definition: ExecutionDefinition,
+        execution: Execution | None = None,
+    ) -> None:
+        """
+        Run a diagnostic in process
+
+        Parameters
+        ----------
+        definition
+            A description of the information needed for this execution of the diagnostic
+        execution
+            A database model representing the execution of the diagnostic.
+            If provided, the result will be updated in the database when completed.
+        """
+        # Submit the execution to the process pool
+        # and track the future so we can wait for it to complete
+        result = _process_run(
+            definition=definition,
+            log_level=self.config.log_level,
+        )
+
+    def join(self) -> None:
+        pass
```

Comment:
Is it possible to track the progress of jobs via the main process?

Reply:
Yes. parsl can track the progress of job and wait for the jobs to complete. I just put an initial piece into the REF. So a lot of functions are missing. I will work on it most time of the incoming week.